numpy/lib/tests/test_nanfunctions.py

```diff
@@ -887,6 +888,134 @@ def test_multiple_percentiles(self):
         megamat = np.ones((3, 4, 5, 6))
         assert_equal(np.nanpercentile(megamat, perc, axis=(1, 2)).shape, (2, 3, 6))
 
+    def test_weights(self):
+        a = np.array([[0., 1., 2.], [3., 4., 5.]])
+        a[0][1] = np.nan
+        # regression tests
+        assert_equal(np.nanpercentile(a, q=50, weights=np.ones(6).reshape(2, 3)),
+                     np.nanpercentile(a, q=50))
+        assert_equal(np.nanpercentile(a, q=50, axis=0, weights=[1, 1]),
+                     np.nanpercentile(a, q=50, axis=0))
+        assert_equal(np.nanpercentile(a, q=50, axis=1, weights=[1, 1, 1]),
+                     np.nanpercentile(a, q=50, axis=1))
+        # unequal weights
+        axis = 1
+        weights = [0, 1, 2]
+        stand_in = np.stack((a[:, 1], a[:, 2], a[:, 2]), axis=axis)
+        assert_equal(np.nanpercentile(a, q=50, axis=axis, weights=weights),
+                     np.nanpercentile(stand_in, q=50, axis=axis))
+
+
+class TestNanFunctions_Quantile(object):
+    # most of this is already tested by TestPercentile
+
+    def test_regression(self):
+        ar = np.arange(24).reshape(2, 3, 4).astype(float)
+        ar[0][1] = np.nan
+
+        assert_equal(np.nanquantile(ar, q=0.5), np.nanpercentile(ar, q=50))
+        assert_equal(np.nanquantile(ar, q=0.5, axis=0),
+                     np.nanpercentile(ar, q=50, axis=0))
+        assert_equal(np.nanquantile(ar, q=0.5, axis=1),
+                     np.nanpercentile(ar, q=50, axis=1))
+        assert_equal(np.nanquantile(ar, q=[0.5], axis=1),
+                     np.nanpercentile(ar, q=[50], axis=1))
+        assert_equal(np.nanquantile(ar, q=[0.25, 0.5, 0.75], axis=1),
+                     np.nanpercentile(ar, q=[25, 50, 75], axis=1))
```

Comment:
This test and some of the others too belong in #10199

Reply:
Done